src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -286,231 +352,229 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
         bool ownerOfAll = true;
-        for (uint256 xi = x; xi < x+3; xi++) {
-            for (uint256 yi = y; yi < y+3; yi++) {
-                ownerOfAll = _checkAndClear(from, xi + yi * GRID_SIZE) && ownerOfAll;
+
+        {
+            for (uint256 xi = land.x; xi < land.x + land.size; xi += childQuadSize) {
+                for (uint256 yi = land.y; yi < land.y + land.size; yi += childQuadSize) {
+                    uint256 ownerChild;
+                    bool ownAllIndividual;
+                    if (childQuadSize < 3) {
+                        ownAllIndividual = _checkAndClear(from, xi + yi * GRID_SIZE) && ownerOfAll;
+                    } else {
+                        ownAllIndividual = _regroupQuad(
+                            from,
+                            to,
+                            Land({x: xi, y: yi, size: childQuadSize}),
+                            false,
+                            childQuadSize / 2
+                        );
+                        uint256 idChild = _getQuadId(childLayer, xi, yi);
+                        ownerChild = _owners[idChild];
+                        if (ownerChild != 0) {
+                            if (!ownAllIndividual) {
+                                require(ownerChild == uint256(from), "not owner of child Quad");
+                            }
+                            _owners[idChild] = 0;
+                        }
+                    }
+                    ownerOfAll = (ownAllIndividual || ownerChild != 0) && ownerOfAll;
+                }
             }
         }
-        if(set) {
-            if(!ownerOfAll) {
-                require(
-                    _ownerOfQuad(3, x, y) == from,
-                    "not owner of all sub quads nor parent quads"
-                );
+
+        if (set) {
+            if (!ownerOfAll) {
+                require(_ownerOfQuad(land.size, land.x, land.y) == from, "not owner of all sub quads nor parent quads");
             }
             _owners[quadId] = uint256(to);
             return true;
         }
+
         return ownerOfAll;
     }
 
-    function _ownerOfQuad(uint256 size, uint256 x, uint256 y) internal view returns (address) {
-        uint256 layer;
-        uint256 parentSize = size * 2;
+    function _idInPath(
+        uint256 i,
+        uint256 size,
+        uint256 x,
+        uint256 y
+    ) internal pure returns (uint256) {
+        uint256 row = i / size;
+        if (row % 2 == 0) {
+            // allow ids to follow a path in a quad
+            return (x + (i % size)) + ((y + row) * GRID_SIZE);
+        } else {
+            return ((x + size) - (1 + (i % size))) + ((y + row) * GRID_SIZE);
+        }
+    }
+
+    function _isQuadMinted(
+        Land[] memory mintedLand,
+        uint256 x,
+        uint256 y,
+        uint256 size,
+        uint256 index
+    ) internal pure returns (bool) {
+        for (uint256 i = 0; i <= index; i++) {
```

Comment:
only less than ?

Reply:
@capedcrusader21

Reply:
Index variable is initialised from zero, That is why comparison is "less than equal to".